calicoctl/commands/node/diags.go

```diff
@@ -155,7 +160,43 @@ such as transfer.sh using curl or similar.  For example:
 
 }
 
-// writeDiags executes the dignostic commans and outputs the result in the file
+// getDockerLogs will attempt to grab logs for any "calico" named containers for hosted installs.
+func getDockerLogs(logDir string) {
+	os.Mkdir(logDir, os.ModeDir)
+
+	fmt.Println("Copying logs from Calico containers")
+	// Get a list of Calico containers running on this Node.
+	var stderr bytes.Buffer
+	cmd := exec.Command("docker", "ps", "-a", "--filter", "\"name=calico\"", "--format", "{{.Names}}")
+	cmd.Stderr = &stderr
```

Comment:
I'm not sure if this is necessary. I think you can just do the following:
```go
out,err := cmd.Output()
if err != nil {
     fmt.Println("blah: %s\n", err.Stderr)
     return
}
```
From Golang spec:
> Any returned error will usually be of type *ExitError. If c.Stderr was nil, Output populates ExitError.Stderr.